parsing/location.mli

```diff
@@ -200,6 +200,7 @@ type report = {
   kind : report_kind;
   main : msg;
   sub : msg list;
+  footnote: unit -> (Format.formatter -> unit) option
```

Comment:
I agree that that the design is weird. What happens is that currently, the printing of the error message is effectful and it might generate new footnotes. Thus we need to delay the computation of the footnote until after the printing. And this design is in fact less weird than the current one, where we might access multiple time to the list of conflicts at printing time and ends up with a printed error message which depends on the evaluation strategy of multiply nested `Format.formatter -> unit` closures.

My plan is to make this weirdness disappears in the next PR which makes the printing of error-message non-effectful without sacrificing formatting.

> (minor) the msg type carries a location, by consistency I would expect the delayed_msg also to locate each footnote. (So: a list of footnotes with each a location, I guess?)

Having a list of footnote with optional locations would be an improvement to the existing error messages, but I will propose to do that later once we don't have anymore delayed footnote producers.